drivers/windows/R/check.R

```diff
@@ -0,0 +1,63 @@
+windows_check <- function() {
+  ok <- windows_check_credentials()
+  ok <- windows_check_connection() && ok
+  invisible(ok)
+}
+
+
+windows_check_credentials <- function() {
+  credentials <- windows_check_credentials_found()
+  if (is.null(credentials)) {
+    FALSE
+  } else {
+    windows_check_credentials_correct(credentials)
+  }
+}
+
+
+windows_check_credentials_found <- function() {
+  credentials <- tryCatch(windows_credentials(), error = identity)
+  if (inherits(credentials, "error")) {
+    cli::cli_alert_danger(credentials$message)
+    NULL
+  } else {
+    cli::cli_alert_success(
+      "Found DIDE credentials for '{credentials$username}'")
+    credentials
+  }
+}
+
+
+windows_check_credentials_correct <- function(credentials) {
+  result <- tryCatch(
+    api_client_login(credentials$username, credentials$password),
+    error = identity)
+  if (inherits(result, "error")) {
+    cli::cli_alert_danger("Failed to log into the portal with your credentials")
+    cli::cli_alert_info(
+      paste("If your password has expired, you will need to reset it",
+            "and then rerun 'hipercow::windows_authenticate()' to",
+            "update the copy in your keyring"),
+      wrap = TRUE)
+    cli::cli_alert_info("Original error message: {result$message}")
+    FALSE
+  } else {
+    cli::cli_alert_success("DIDE credentials are correct")
+    TRUE
+  }
+}
+
+
+windows_check_connection <- function(timeout = 1) {
+  result <- tryCatch({
+    httr::HEAD("https://vault.dide.ic.ac.uk:8200", httr::timeout(timeout))
+  }, error = identity)
+  if (inherits(result, "error")) {
+    cli::cli_alert_danger("Failed to make connection with private network")
+    cli::cli_alert_info("Please check that you have ZScaler enabled")
```

Comment:
We could say here...
* Check you have working internet access (we could check for this by pinging 8.8.8.8 or something)
* If you are connecting from outside DIDE, check you have ZScaler enabled

A little faffy - perhaps future PRs. What's here already is the likeliest scenario...